packages/block-editor/src/components/link-control/index.js

```diff
@@ -134,6 +141,19 @@ function LinkControl( {
 	hasTextControl = false,
 	renderControlBottom = null,
 } ) {
+	if ( hasNofollowSetting ) {
```

Comment:
I introduced this `hasNoFollowSetting` so that LinkControl can render the `nofollow` checkbox based on the assigned prop, this can help reduce the default overridden on various places, eg [inline.js](https://github.com/WordPress/gutenberg/blob/ab5a41c7979b81f0827289f1adde7f505bcc2702/packages/format-library/src/link/inline.js#L34)
Please let me know what you think @getdave?

Reply:
I'm not sure we need/want to do this.

The API of this component is already extremely large and continuing to add to it only compounds the confusion.

We already have a means of determining whether or not to render settings fields. I would prefer instead if we perhaps considering exposing some other standardised settings as part of `LinkControl` in a simple fashion as we do for [the default settings](https://github.com/WordPress/gutenberg/blob/ab5a41c7979b81f0827289f1adde7f505bcc2702/packages/format-library/src/link/inline.js#L35).

That way consumers could optionally utilise those standard settings without using having to further complicate the API of this component.

My thinking is influecned by the thought that whilst `nofollow` might relevant _today_ it could easily not be in future. If we make it a standard of the component's API then we're committing to maintaining it forever. What happens if Google says "we no longer recommend this" and then we've added this prop? I'd rather avoid that.

Whilst I'm conscious that my proposal is perhaps less flexible and requires more wiring, it helps to limit the surface area of backwards compatibility issues.

What do you think?

Reply:
The points make sense to me 👍 
So for now we just follow the way `inline.js` is using to append new settings to the link

Reply:
That sounds good to me for now. We can always refactor later.

Reply:
Addressed in latest commit https://github.com/WordPress/gutenberg/pull/54110/commits/07cee9c9dea8b765070bd2ca4665c7a91c0c4fa0